### PR TITLE
Improve metadata shutdown responsiveness.

### PR DESCRIPTION
**Pull request by DrTall**
cc @zorkian 

In practice `MetadataRefreshFrequency` values tend to be large so this original setup was very bad.

### CODE REVIEW

File: connection_pool.go [hunk 1]
```diff
@@ -359,10 +361,20 @@ func (cp *connectionPool) Close() {
 		backend.Close()
 	}
 	cp.backends = nil
+	if !cp.closed {
+		close(cp.closedChan)
```
Comment (zorkian) [inline connection_pool.go:365]: Did we ever decide on whether or not it's safe to close a channel while other people are reading against it? It's certainly safe to read from a closed channel, but we received reports of races detected when closing a channel we're reading from immediately.

**Review by zorkian — APPROVED** [review]
LGTM feel free to merge if you don't think my comment warrants any changes (I don't remember where we left off on this issue)

File: connection_pool.go (hunk 1 above)
Comment (DrTall) [reply connection_pool.go:365]: It is safe. The race that Ping encountered was attempting to read from a channel that was being created in another goroutine. That is, the race is actually on the struct that owns the channel object, not the channel itself.

Comment (zorkian) [reply connection_pool.go:365]: Ping's not the only one that saw it, but alright. LGTM.

**Pull request merged**